src/range_proof.rs

```diff
@@ -750,28 +750,26 @@ where
 
     // Helper function to decompress Li
     fn li_decompressed(&self) -> Result<Vec<P>, ProofError> {
-        let mut li = Vec::with_capacity(self.li.len());
-        for item in &self.li {
-            li.push(item.decompress().ok_or_else(|| {
-                ProofError::InvalidArgument(
+        self.li
+            .iter()
+            .map(|p| {
+                p.decompress().ok_or(ProofError::InvalidArgument(
                     "An item in member 'L' was not the canonical encoding of a point".to_string(),
-                )
-            })?)
-        }
-        Ok(li)
+                ))
+            })
+            .collect::<Result<Vec<_>, ProofError>>()
```

Comment:
Not at all NB but specifying the types here is redundant because the return type already specifies the type you are collecting to, so a `.collect()` works fine.  Same for below.

Reply:
Updated in a new commit!